operator/pkg/awsprovider/launchtemplate/reconciler.go

```diff
@@ -133,6 +135,28 @@ func (c *Controller) createLaunchTemplate(ctx context.Context, dataplane *v1alph
 	return nil
 }
 
+func (c *Controller) amiID(ctx context.Context, dataplane *v1alpha1.DataPlane) (string, error) {
+	kubeVersion, err := c.desiredKubernetesVersion(ctx, dataplane)
+	if err != nil {
+		return "", fmt.Errorf("getting kubernetes version, %w", err)
+	}
+	paramOutput, err := c.ssm.GetParameterWithContext(ctx, &ssm.GetParameterInput{
```

Comment:
can this be picked from spec instead of determining from CP k8s version ? 
We might want to test different patch version upto ~ 2 patch versions back from current CP version.